Dockerfile.nccl

```diff
@@ -0,0 +1,87 @@
+# Start with the NVIDIA CUDA base image
+FROM nvidia/cuda:12.2.0-devel-ubuntu22.04@sha256:7e81c70fb55bfc54e6711700eb721bb94159a3cd546620b37210d12c17914fa6
```

Comment:
see if we need higher cuda version 
The sha was added previously because I build it in my local Mac and the arch is not compatible with the test environment. So please check what arch is in the image build env and it would better to remove the sha at the end